sql3/planner/expressionpql.go

```diff
@@ -96,6 +96,85 @@ func (p *ExecutionPlanner) generatePQLCallFromExpr(ctx context.Context, expr typ
 			}
 			return call, nil
 
+		case "RANGEQ":
+			col := expr.args[0].(*qualifiedRefPlanExpression)
+
+			var fromValue interface{}
+			switch fromExpr := expr.args[1].(type) {
+			case *stringLiteralPlanExpression:
+				// parse timestamp from string and use the int value
+				ts := fromExpr.ConvertToTimestamp()
+				if ts == nil {
+					return nil, sql3.NewErrInvalidTypeCoercion(0, 0, fromExpr.value, parser.NewDataTypeTimestamp().TypeDescription())
+				}
+				fromValue = ts.Unix()
+
+			case *intLiteralPlanExpression:
+				// use the int value
+				fromValue = fromExpr.value
+
+			case *nullLiteralPlanExpression:
+				fromValue = nil
+
+			default:
+				return nil, sql3.NewErrInternalf("unexpected argument type '%T'", expr.args[1])
+			}
+
+			var toValue interface{}
+			switch toExpr := expr.args[2].(type) {
+			case *stringLiteralPlanExpression:
+				// parse timestamp from string and use the int value
+				ts := toExpr.ConvertToTimestamp()
+				if ts == nil {
+					return nil, sql3.NewErrInvalidTypeCoercion(0, 0, toExpr.value, parser.NewDataTypeTimestamp().TypeDescription())
+				}
+				toValue = ts.Unix()
+
+			case *intLiteralPlanExpression:
+				// use the int value
+				toValue = toExpr.value
+
+			case *nullLiteralPlanExpression:
+				toValue = nil
+
+			default:
+				return nil, sql3.NewErrInternalf("unexpected argument type '%T'", expr.args[1])
+			}
+
+			if fromValue == nil && toValue == nil {
+				return nil, sql3.NewErrInternalf("replace me")
```

Comment:
I think this wants to be replaced but I'm not sure what with. I assume the reason this is an Internalf is that we shouldn't be able to get here without having hit an earlier error, possibly in parsing?

Reply:
Good catch - fixed in subsequent commit